ibis/backends/__init__.py

```diff
@@ -805,12 +806,7 @@ def __init__(self, *args, **kwargs):
         self._con_args: tuple[Any] = args
         self._con_kwargs: dict[str, Any] = kwargs
         self._can_reconnect: bool = True
-        # expression cache
-        self._query_cache = RefCountedCache(
-            populate=self._load_into_cache,
-            lookup=lambda name: self.table(name).op(),
-            finalize=self._clean_up_cached_table,
-        )
+        self._query_cache = QueryCache(weakref.proxy(self))
```

Comment:
Actually, the `weakref` here is pointless, because `ops.DatabaseTable`s produced by the `Table.cache()` method hold a strong reference to the backend.

While I don't like that tables can outlive the backend, I'm not it's reasonable to artificially disable that without a strong motivation.

The cleanups are still beneficial here, so I'll keep this PR open and cleanup the title and description and remove the `weakref`.